CSharp/ArticleProjects/PersistOutputs/FileConventionsExample.cs

```diff
@@ -0,0 +1,107 @@
+﻿namespace Microsoft.Azure.Batch.Samples.Articles.PersistOutputs
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using WindowsAzure.Storage;
+    using WindowsAzure.Storage.Blob;
+    using Common;
+    using Conventions.Files;
+
+    public static class FileConventionsExample
+    {
+        public static CloudBlobContainer Run(
+            BatchClient batchClient,
+            CloudStorageAccount linkedStorageAccount,
+            string poolId,
+            int nodeCount,
+            string jobId)
+        {
+            const string appPackageId = "PersistOutputsTask";
+            const string appPackageVersion = "1.0";
+
+            // Create and configure an unbound pool.
+            CloudPool pool = batchClient.PoolOperations.CreatePool(poolId: poolId,
+                virtualMachineSize: "small",
+                targetDedicatedComputeNodes: nodeCount,
+                cloudServiceConfiguration: new CloudServiceConfiguration(osFamily: "4"));
+
+            // Specify the application and version to deploy to the compute nodes. You must
+            // first build PersistOutputsTask, then upload it as an application package.
+            // See https://azure.microsoft.com/documentation/articles/batch-application-packages/
+            pool.ApplicationPackageReferences = new List<ApplicationPackageReference>
+            {
+                new ApplicationPackageReference
+                {
+                    ApplicationId = appPackageId,
+                    Version = appPackageVersion
+                }
+            };
+
+            // Commit the pool to the Batch service
+            GettingStartedCommon.CreatePoolIfNotExistAsync(batchClient, pool).Wait();
+
+            CloudJob job = batchClient.JobOperations.CreateJob(jobId, new PoolInformation { PoolId = poolId });
+
+            // Create the blob storage container for the outputs.
+            job.PrepareOutputStorageAsync(linkedStorageAccount).Wait();
+
+            // Create an environment variable on the compute nodes that the
+            // task application can reference when persisting its outputs.
+            string containerName = job.OutputStorageContainerName();
+            CloudBlobContainer container = linkedStorageAccount.CreateCloudBlobClient().GetContainerReference(containerName);
+            string containerSas = container.GetSharedAccessSignature(CreateFullAccessPolicy());
+            string containerUrl = container.Uri.AbsoluteUri + containerSas;
```

Comment:
Is there a particular reason not to use the built-in `job.GetOutputStorageContainerUrl()`?  It seems to do exactly the same thing except for a different access policy - and if the reason for doing this in app code is to control the access policy then let's just add a parameterised overload to GOSCU.